pkg/virt-controller/watch/updatereactor_test.go

```diff
@@ -0,0 +1,86 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 The KubeVirt Authors
+ *
+ */
+package watch
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/testing"
+	v1 "kubevirt.io/api/core/v1"
+)
+
+func SubresourceHandle(subresources string) bool {
+	return subresources == "status"
+}
+
+func Handle(subresources string) bool {
+	return subresources != "status"
+}
+
+func ModifyStatusOnlyVM(new, old runtime.Object) runtime.Object {
+	vm := new.(*v1.VirtualMachine)
+	oldVM := old.(*v1.VirtualMachine)
```

Comment:
Can be these 2 object nil? Should we add a check?

Reply:
afaik they cannot be nil